daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -169,29 +200,40 @@ object Runner {
       // Map is but it doesn’t return a Map so we have to call toMap afterwards.
       defaultClient <- Future
         .traverse(participantParams.default_participant.toList)(x =>
-          connectApiParameters(x, clientConfig, maxInboundMessageSize))
+          connectApiParameters(x, applicationId, tlsConfig, maxInboundMessageSize))
         .map(_.headOption)
       participantClients <- Future
         .traverse(participantParams.participants: Map[Participant, ApiParameters])({
-          case (k, v) => connectApiParameters(v, clientConfig, maxInboundMessageSize).map((k, _))
+          case (k, v) =>
+            connectApiParameters(v, applicationId, tlsConfig, maxInboundMessageSize).map((k, _))
         })
         .map(_.toMap)
     } yield Participants(defaultClient, participantClients, participantParams.party_participants)
   }
 
-  def jsonClients(
-      participantParams: Participants[ApiParameters],
-      token: String,
-      envIface: EnvironmentInterface)(
+  def jsonClients(participantParams: Participants[ApiParameters], envIface: EnvironmentInterface)(
       implicit ec: ExecutionContext,
       system: ActorSystem): Future[Participants[JsonLedgerClient]] = {
     def client(params: ApiParameters) = {
       val uri = Uri(params.host + ":" + params.port.toString)
-      new JsonLedgerClient(uri, Jwt(token), envIface, system)
+      params.access_token match {
+        case None =>
+          Future.failed(new RuntimeException(s"The JSON API always requires access tokens"))
+        case Some(token) =>
+          Future.successful(new JsonLedgerClient(uri, Jwt(token), envIface, system))
+      }
+
     }
-    val defClient = participantParams.default_participant.map(client(_))
-    val otherClients = participantParams.participants.map({ case (k, v) => (k, client(v)) })
-    Future { Participants(defClient, otherClients, participantParams.party_participants) }
+    for {
+      // Apparently Future.traverse is too stupid to traverse an
+      // Option so you first have to convert to a list and then convert back.
+      defClient <- Future
+        .traverse(participantParams.default_participant.toList)(client(_))
+        .map(_.headOption)
```

Comment:
Maybe scalaz has something for this? (Since it's imported already anyway)

Reply:
I looked for it but scalaz doesn’t make `Future` an applicative so it doesn’t help either.